app/src/main/java/com/example/healthtracker/ExerciseDiaryActivity.java

```diff
@@ -2,38 +2,78 @@
 
 import android.os.Bundle;
 import android.view.View;
-import android.widget.TextView;
+import android.widget.Button;
+import android.widget.EditText;
 
+import java.text.SimpleDateFormat;
 import java.util.Date;
+import java.util.List;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 import androidx.room.Room;
 
 public class ExerciseDiaryActivity extends AppCompatActivity {
 
-    TextView exerciseText;
+    RecyclerView.LayoutManager layoutManager;
+    RecyclerView recyclerView;
+    RecyclerView.Adapter adapter;
+
+    Button button;
+
+    ExerciseDatabase db;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_exercise_diary);
 
-        ExerciseDatabase db = Room.databaseBuilder(getApplicationContext(), ExerciseDatabase.class,"exercise-db")
+
+        db = Room.databaseBuilder(getApplicationContext(), ExerciseDatabase.class,"exercise-db")
                .allowMainThreadQueries()
                 .build();
 
         Date date = new Date();
         long time = date.getTime();
 
 
-        Exercise ToeStrength = new Exercise("Strong Toe", 5, "Core toe strength is the most under utilized exercise around. Have strong toesies!", date);
+        List<Exercise> exercises = db.exerciseDao().getAll();
+
+
+
+        RecyclerView recyclerView = findViewById(R.id.exerciseText);
+        layoutManager = new LinearLayoutManager(this);
+        recyclerView.setLayoutManager(layoutManager);
+
+        adapter = new ExerciseLayoutAdapter(exercises);
+        recyclerView.setAdapter(adapter);
+
+    }
+
+    public void submitButtonClick(View view) throws Exception{
+
+        EditText titleInput = findViewById(R.id.title);
+        EditText quantityInput = findViewById(R.id.quantity);
+        EditText descriptionInput = findViewById(R.id.description);
+        EditText dateInput = findViewById(R.id.date);
+
+        String title = titleInput.getText().toString();
+        String quantity = quantityInput.getText().toString();

```

Comment:
This could be put in as a string which would break your app on line 65. Change this input type to a number.